aws/data_source_aws_prefix_list.go

```diff
@@ -45,31 +46,34 @@ func dataSourceAwsPrefixListRead(d *schema.ResourceData, meta interface{}) error
 	if prefixListID := d.Get("prefix_list_id"); prefixListID != "" {
 		req.PrefixListIds = aws.StringSlice([]string{prefixListID.(string)})
 	}
-	req.Filters = buildEC2AttributeFilterList(
-		map[string]string{
-			"prefix-list-name": d.Get("name").(string),
-		},
-	)
+	if prefixListName := d.Get("name"); prefixListName.(string) != "" {
+		req.Filters = append(req.Filters, &ec2.Filter{
+			Name:   aws.String("prefix-list-name"),
+			Values: aws.StringSlice([]string{prefixListName.(string)}),
+		})
+	}
 
 	log.Printf("[DEBUG] Reading Prefix List: %s", req)
 	resp, err := conn.DescribePrefixLists(req)
-	if err != nil {
+	switch {
+	case err != nil:
 		return err
-	}
-	if resp == nil || len(resp.PrefixLists) == 0 {
+	case resp == nil || len(resp.PrefixLists) == 0:
 		return fmt.Errorf("no matching prefix list found; the prefix list ID or name may be invalid or not exist in the current region")
+	case len(resp.PrefixLists) > 1:
+		return fmt.Errorf("more than one prefix list matched the given set of criteria")
 	}
 
 	pl := resp.PrefixLists[0]
 
 	d.SetId(*pl.PrefixListId)
 	d.Set("name", pl.PrefixListName)
 
-	cidrs := make([]string, len(pl.Cidrs))
-	for i, v := range pl.Cidrs {
-		cidrs[i] = *v
+	cidrs := aws.StringValueSlice(pl.Cidrs)
+	sort.Strings(cidrs)
```

Comment:
Similarly here with respect to introducing a breaking change which should be removed for now.

In this case though, if the ordering of the CIDR Blocks is not significant (which it likely is not), the `cidr_blocks` attribute should be changed to `TypeSet` instead of attempting to add ordering that is inconsistent with the semantics of the upstream API. Terraform configurations can and should use built in language support such as resource `for_each` for handling all elements of an attribute like this to avoid ordering concerns.

At some point in the future we may go back through many of the `TypeList` attributes and convert them to `TypeSet` to ensure that Terraform configurations are not depending on ordering where its not guaranteed or semantically correct.